dockerclient/client.go

```diff
@@ -833,6 +833,45 @@ func snapshotPath(path, containerID string, client *docker.Client) (string, erro
 	}
 	glog.V(4).Infof("Snapshot %s for later use under %s", path, f.Name())
 
+	ensurePath := func(dest string) error {
+		if !strings.HasSuffix(dest, "/") {
+			dest = dest + "/"
+		}
+		glog.V(4).Infof("Running mkdir -p %q", dest)
+		exec, err := client.CreateExec(docker.CreateExecOptions{
+			Container: containerID,
+			Cmd:       []string{"/bin/sh", "-c", "mkdir -p $@", "", dest},
```

Comment:
This will fail on images that don't have SH like scratch images.  Won't a tar file successfully create these?

Reply:
It won't be created as a side-effect if the layer doesn't include any content to put inside of it.

Reply:
This makes images without SH unbuildable though - so we can't do this as is.  What other options do we have?

Reply:
Okay, uploading an empty directory seems to work, and doesn't require us to exec anything.  I've updated the patch to try doing it that way.